src/topology/instances/add_circle.lean

```diff
@@ -126,6 +154,86 @@ instance : divisible_by (add_circle p) ℤ :=
     exact (equiv_Ico p).symm_apply_apply x,
   end, }
 
+end floor_ring
+
+section finite_order_points
+
+variables {p}
+
+/-- See also `add_circle.gcd_mul_add_order_of_div_eq`. -/
+lemma add_order_of_div_of_gcd_eq_one {m n : ℕ} (hn : 0 < n) (h : gcd n m = 1) :
+  add_order_of (↑(↑m / ↑n * p) : add_circle p) = n :=
```

Comment:
Would it be useful to have a version of this for `ℚ` as well? Or is the problem that there is no coercion from `Q` to `add_circle p`?

Reply:
A version for `ℚ` is a good idea. I have just added the lemmas `add_order_of_div_of_gcd_eq_one'` and `add_order_of_coe_rat` to cater for this.

Since you mention it, there is no coercion from `ℚ` because it would only be useful if it also multiplied by the period `p` but then it would disagree with the real version (unless `p = 1`). So for now at least I'm just not defining the coercion.